crates/daphne/src/protocol/client.rs

```diff
@@ -65,17 +68,28 @@ impl VdafConfig {
         time: Time,
         task_id: &TaskId,
         report_id: &ReportId,
-        extensions: Vec<Extension>,
+        public_extensions: Option<Vec<Extension>>,
+        private_extensions: Vec<Extension>,
         version: DapVersion,
     ) -> Result<Report, DapError> {
+        match (&public_extensions, version) {
+            (Some(_), DapVersion::Draft09) | (None, DapVersion::Latest) => {
+                return Err(DapError::ReportError(
+                    crate::messages::ReportError::InvalidMessage,
+                ))
```

Comment:
I would actually go with a fatal error here
```suggestion
                return Err(fatal_error!(
                    err = format!("public extensions not set correctly for {version:?}")
                ));
```